Source/MonoGame.Extended/Transform.cs

```diff
@@ -0,0 +1,281 @@
+                parent = parent.ParentTransform;
+            }
+        }
+
+        private void ParentTransformOnTransformBecameDirty()
+        {
+            _flags |= TransformFlags.WorldMatrixIsDirty;
+        }
+
+        private void RecalculateWorldMatrixIfNecessary()
+        {
+            if ((_flags & TransformFlags.WorldMatrixIsDirty) == 0)
+                return;
+
+            RecalculateLocalMatrixIfNecessary();
+            RecalculateWorldMatrix(ref _localMatrix, out _worldMatrix);
+            _flags &= ~TransformFlags.WorldMatrixIsDirty;
+            TranformUpdated?.Invoke();
+        }
+
+        protected abstract void RecalculateWorldMatrix(ref TMatrix localMatrix, out TMatrix matrix);
+
+        private void RecalculateLocalMatrixIfNecessary()
+        {
+            if ((_flags & TransformFlags.LocalMatrixIsDirty) == 0)
+                return;
+
+            RecalculateLocalMatrix(out _localMatrix);
+
+            _flags &= ~TransformFlags.LocalMatrixIsDirty;
+            WorldMatrixBecameDirty();
+        }
+
+        protected abstract void RecalculateLocalMatrix(out TMatrix matrix);
+    }
+
+    /// <summary>
+    ///     Represents the position, rotation, and scale of a two-dimensional game object.
+    /// </summary>
+    /// <seealso cref="Extended.BaseTransform{Matrix2D, Transform2D}" />
+    /// <remarks>
+    ///     <para>
+    ///         Every game object has a transform which is used to store and manipulate the position, rotation and scale
+    ///         of the object. Every transform can have a parent, which allows to apply position, rotation and scale to game
+    ///         objects hierarchically.
+    ///     </para>
+    /// </remarks>
+    public class Transform2D : BaseTransform<Matrix2D, Transform2D>
+    {
+        private Vector2 _position;
+        private Vector2 _scale = Vector2.One;
+        private float _rotationAngle;
+
+        /// <summary>
+        ///     Gets or sets the position.
+        /// </summary>
+        /// <value>
+        ///     The position.
+        /// </value>
+        public Vector2 Position
+        {
+            get { return _position; }
+            set
+            {
+                _position = value;
+                LocalMatrixBecameDirty();
+                WorldMatrixBecameDirty();
+            }
+        }
+
+        /// <summary>
+        ///     Gets or sets the scale.
+        /// </summary>
+        /// <value>
+        ///     The scale.
+        /// </value>
+        public Vector2 Scale
+        {
+            get { return _scale; }
+            set
+            {
+                _scale = value;
+                LocalMatrixBecameDirty();
+                WorldMatrixBecameDirty();
+            }
+        }
+
+        /// <summary>
+        ///     Gets or sets the rotation angle.
+        /// </summary>
+        /// <value>
+        ///     The rotation angle.
+        /// </value>
+        public float RotationAngle
```

Comment:
I think we should just call this `Rotation` to be consistent with other parts of the framework. Also, since you've got documentation, we should probably mention that it's in radians.